sdk/python/tests/test_client.py

```diff
@@ -495,516 +534,100 @@ def test_list_feature_sets(self, mocked_client, mocker):
-    ):
-
-        test_client.set_project("project1")
-        driver_fs = FeatureSet(
-            "driver-feature-set",
-            source=KafkaSource(brokers="localhost:4412", topic="test"),
-        )
-        driver_fs.add(Feature(name="feature_1", dtype=ValueType.FLOAT))
-        driver_fs.add(Feature(name="feature_2", dtype=ValueType.STRING))
-        driver_fs.add(Feature(name="feature_3", dtype=ValueType.INT64))
-        driver_fs.add(Entity(name="entity_id", dtype=ValueType.INT64))
-
-        # Register with Feast core
-        test_client.apply(driver_fs)
-        driver_fs = driver_fs.to_proto()
-        driver_fs.meta.status = FeatureSetStatusProto.STATUS_READY
-
-        mocker.patch.object(
-            test_client._core_service_stub,
-            "GetFeatureSet",
-            return_value=GetFeatureSetResponse(feature_set=driver_fs),
-        )
-
-        with pytest.raises(exception):
-            test_client.ingest("driver-feature-set", dataframe, timeout=1)
-
-    @pytest.mark.parametrize(
-        "dataframe,exception,test_client",
-        [
-            (dataframes.GOOD, TimeoutError, lazy_fixture("client")),
-            (dataframes.GOOD, TimeoutError, lazy_fixture("secure_client")),
-        ],
-    )
-    def test_feature_set_ingest_fail_if_pending(
-        self, dataframe, exception, test_client, mocker
-    ):
-        with pytest.raises(exception):
-            test_client.set_project("project1")
-            driver_fs = FeatureSet(
-                "driver-feature-set",
-                source=KafkaSource(brokers="kafka:9092", topic="test"),
-            )
-            driver_fs.add(Feature(name="feature_1", dtype=ValueType.FLOAT))
-            driver_fs.add(Feature(name="feature_2", dtype=ValueType.STRING))
-            driver_fs.add(Feature(name="feature_3", dtype=ValueType.INT64))
-            driver_fs.add(Entity(name="entity_id", dtype=ValueType.INT64))
-
-            # Register with Feast core
-            test_client.apply(driver_fs)
-            driver_fs = driver_fs.to_proto()
-            driver_fs.meta.status = FeatureSetStatusProto.STATUS_PENDING
-
-            mocker.patch.object(
-                test_client._core_service_stub,
-                "GetFeatureSet",
-                return_value=GetFeatureSetResponse(feature_set=driver_fs),
-            )
-
-            # Need to create a mock producer
-            with patch("feast.client.get_producer"):
-                # Ingest data into Feast
-                test_client.ingest("driver-feature-set", dataframe, timeout=1)
-
-    @pytest.mark.parametrize(
-        "dataframe,exception,test_client",
-        [
-            (dataframes.BAD_NO_DATETIME, Exception, lazy_fixture("client")),
-            (
-                dataframes.BAD_INCORRECT_DATETIME_TYPE,
-                Exception,
-                lazy_fixture("client"),
+        # Create Feature Tables
+        batch_source = DataSource(
+            type="BATCH_FILE",
+            field_mapping={
```

Comment:
What is the point of this field mapping?

Reply:
Added as field mapping is compulsory parameter.